spynnaker/pyNN/models/neural_projections/connectors/fixed_number_pre_connector.py

```diff
@@ -36,69 +35,168 @@ def get_delay_maximum(self):
+
+                # If verbose then output the list connected to this post-neuron
+                if self._verbose:
+                    print 'post-neuron ', m, ' connects to pre-neurons '
+                    print self._pre_neurons[m]
+                    numpy.savetxt(file_handle,
+                                  self._pre_neurons[m][None, :],
+                                  fmt=("%u,"*(self._n_pre-1)+"%u"))
+
         return self._pre_neurons
 
-    def _pre_neurons_in_slice(self, pre_vertex_slice):
+    def _pre_neurons_in_slice(self, pre_vertex_slice, n):
         pre_neurons = self._get_pre_neurons()
-        return pre_neurons[
-            (pre_neurons >= pre_vertex_slice.lo_atom) &
-            (pre_neurons <= pre_vertex_slice.hi_atom)]
 
-    def _is_connected(self, pre_vertex_slice):
-        return self._pre_neurons_in_slice(pre_vertex_slice).size > 0
+        # Take the nth array and get the bits from it we need
+        # for this pre-vertex slice
+        this_pre_neuron_array = pre_neurons[n]
+
+        return this_pre_neuron_array[
+            (this_pre_neuron_array >= pre_vertex_slice.lo_atom) &
+            (this_pre_neuron_array <= pre_vertex_slice.hi_atom)]
+
+    def _is_connected(self, pre_vertex_slice, n):
+        return self._pre_neurons_in_slice(pre_vertex_slice, n).size > 0
 
     def get_n_connections_from_pre_vertex_maximum(
             self, pre_slices, pre_slice_index, post_slices,
             post_slice_index, pre_vertex_slice, post_vertex_slice,
             min_delay=None, max_delay=None):
 
-        if not self._is_connected(pre_vertex_slice):
+        if not self._is_connected(pre_vertex_slice, 0):
             return 0
 
+        # the number of max connections is either n_pre
+        # or post_vertex_slice.n_atoms
+        n_connections = 0
+        if self._n_pre > post_vertex_slice.n_atoms:
+            n_connections = self._n_pre
+        else:
+            n_connections = post_vertex_slice.n_atoms
+
         if min_delay is None or max_delay is None:
-            return post_vertex_slice.n_atoms
+            return n_connections  # self._n_pre  # post_vertex_slice.n_atoms
 
         return self._get_n_connections_from_pre_vertex_with_delay_maximum(
             self._delays, self._n_pre * self._n_post_neurons,
-            post_vertex_slice.n_atoms, None, min_delay, max_delay)
+            n_connections, None, min_delay, max_delay)
 
     def get_n_connections_to_post_vertex_maximum(
             self, pre_slices, pre_slice_index, post_slices,
             post_slice_index, pre_vertex_slice, post_vertex_slice):
-        if not self._is_connected(pre_vertex_slice):
+        if not self._is_connected(pre_vertex_slice, 0):
             return 0
-        return self._n_pre
+
+        n_connections = 0
+        lo = post_vertex_slice.lo_atom
+        hi = post_vertex_slice.hi_atom
+        for n in range(0, self._n_post_neurons):
+            if (n >= lo and n <= hi):
```

Comment:
Change to loop over range(lo, hi + 1)

Reply:
Looks like this is still there...

Reply:
Oops - missed that one somehow.  Found another duplicated set of lines later on as well... fixed now.